src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp

```diff
@@ -229,22 +205,30 @@ class DiscoveryDataBase
     static eprosima::fastrtps::rtps::GUID_t guid_from_change(
             const eprosima::fastrtps::rtps::CacheChange_t* ch);
 
+    ////////////
+    // Functions to work with own server
+
+    // return the cache change of the server
     fastrtps::rtps::CacheChange_t* cache_change_own_participant();
 
     const std::vector<fastrtps::rtps::GuidPrefix_t> direct_clients_and_servers();
 
     fastrtps::rtps::LocatorList_t participant_metatraffic_locators(
             fastrtps::rtps::GuidPrefix_t participant_guid_prefix);
+            
+    // return a list of participants that are not the server one
+    const std::vector<fastrtps::rtps::GuidPrefix_t> remote_participants();
 
+    // set if the server has been acked by all the reader proxys in server
     bool server_acked_by_all() const
     {
-        return server_acked_by_all_;
+        return server_acked_by_all_.load();
     }
 
     void server_acked_by_all(
             bool s)
     {
-        server_acked_by_all_ = s;
+        server_acked_by_all_.store(s);
     }
```

Comment:
These were already atomic operations. The change is for valgrind sake?

Reply:
I thought it would be nice to use atomic operations explicitly.

Reply:
fine.